### DISCUSSION TITLE
Adding an ingestion mechanism

### DISCUSSION THREAD

**Discussion by userbox020** [turboderp-org/exui#12] [General]
Sup bro, how can we input repos to the llm context so we can do something like chat with a repo feature?

**Comment by turboderp** [comment]
RAG is a very big feature that requires a whole framework with an embedding database on the back end along with a retrieval and prompt stuffing mechanism. It needs to be worked into the conversation format, and to work well it really calls for models that are specifically finetuned for retrieval. So it's a little outside the scope of what I can currently manage on my own. PRs are welcome, though, and I can help with integration into the UI and so on.

**Comment by userbox020** [comment]
you the boss bro, say where to start lets do small steps but solid ones
